test/test_cpp_api_parity.py

```diff
@@ -46,7 +46,7 @@
   } else if (ivalue1.isBool()) {
     return ivalue1.toBool() == ivalue2.toBool();
   } else if (ivalue1.isString()) {
-    return ivalue1.toString() == ivalue2.toString();
+    return ivalue1.toStringRef() == ivalue2.toStringRef();
```

Comment:
What happened here? Does string ref equality check that the pointers are equal or that the strings are equal?

Reply:
`IValue.toString()` returns `c10::intrusive_ptr<ivalue::ConstantString>` and its equality check doesn't check for string content equality. `IValue.toStringRef()` returns `const std::string&`, and its equality check checks that the strings are equal.

Reply:
Interesting... the equality behaivior on `c10::intrusive_ptr<ivalue::ConstantString>` feels like a bug but that is tangential